src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoadContext.cs

```diff
@@ -231,15 +231,37 @@ private Assembly LoadRuntimeAssembly(AssemblyName assemblyName, ScriptRuntimeAss
                 return LoadCore(assemblyName);
             }
 
+            AssemblyName resolvedName = assemblyName;
```

Comment:
As discussed, let's move the policy evaluation up so that we return the deps assembly if the evaluation fails.